### PR TITLE
Closes #1333: Implementing onCrash in Gecko Release/nightly/beta

### LINKED ISSUE #1333: (lib-crash) After simulating a non fatal crash ("about:crashcontent")  on GV and clicking return to the app gets a black screen
This [video](https://drive.google.com/file/d/1vlsCtP4CVadFEI06Mgspt2OAjji3Cy80/view) shows the steps to replicate the issue

### PR DESCRIPTION
**Pull request by Amejia481**

### CODE REVIEW

File: docs/changelog.md [hunk 1]
```diff
@@ -28,6 +28,9 @@ permalink: /changelog/
 * **lib-fetch-okhttp**
   * 🆕 New component: `concept-fetch` implementation using [OkHttp](https://github.com/square/okhttp).
 
+* **engine-gecko,engine-gecko-beta and engine-gecko-nightly**
+  * Fixing bug #1333. This issue didn't allow to use a `GeckoEngineSession` after sending a crash report.
+
```
Comment (pocmo) [inline docs/changelog.md]: This needs to be moved to 0.33.0 now :)

**Comment by pocmo** [comment]
@csadilek Can you review this PR? :)

File: docs/changelog.md (hunk 1 above)
Comment (Amejia481) [reply docs/changelog.md]: 👍

**Comment by Amejia481** [comment]
Sure, I'll rename it! 👍

**Comment by Amejia481** [comment]
Sure, I'll rename it 👍 

It said that but I renamed it, because I was unsure which one to pick 😅 
image

File: components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt [hunk 2]
```diff
@@ -483,6 +475,22 @@ class GeckoEngineSession(
         return null
     }
 
+    internal fun initGeckoSession() {
```
Comment (csadilek) [inline components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt]: Looks like this method can be private?

File: components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt [hunk 3]
```diff
@@ -482,6 +474,22 @@ class GeckoEngineSession(
         return null
     }
 
+    internal fun initGeckoSession() {
```
Comment (csadilek) [inline components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt]: private :) ?

File: components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt [hunk 4]
```diff
@@ -485,6 +477,22 @@ class GeckoEngineSession(
         return null
     }
 
+    internal fun initGeckoSession(geckoSession: GeckoSession = GeckoSession()) {
```
Comment (csadilek) [inline components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt]: private :) ?

File: components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt (hunk 2 above)
Comment (Amejia481) [reply components/browser/engine-gecko-beta/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt]: 👍

**Pull request merged**